backend/app/services/observability/PostgresClient.scala

```diff
@@ -193,4 +201,30 @@ class PostgresClientImpl(postgresConfig: PostgresConfig) extends PostgresClient
             case Failure(exception) => Left(PostgresReadFailure(exception, s"getEvents failed: ${exception.getMessage}"))
         }
     }
+
+  def deleteBlobIngestionEvents(blobId: String): Either[GiantFailure, Long] = {
+    Try {
+      sql"""
+        DELETE FROM ingestion_events WHERE blob_id = ${blobId}
```

Comment:
That's a good point. Right now there is a chance that the first delete operation succeeds but then the second one fails. Meaning, blob events are deleted from ingestion_events table, but the metadata stays there in metadata table.  Using transaction eliminates this, right?

Reply:
I'm missing a lot of context but looking at the [migration](https://github.com/guardian/giant/blob/main/postgres/migrate-db/src/migrations/001.do.create_table_ingestion_events.sql) it's not clear to me why there's no primary key in either table.
Wouldn't making `ingest_id` the primary key in `ingestion_events` and having it referenced in `blob_metadata` with a foreign key relation + `on delete cascade` solve your problem upstream?
Remembering to do things transactionally feels like an added burden to the developer which could be alleviated by enforcing constraints

Reply:
I think the reason we're not using a primary key is because basically any row could be duplicated - if a blob was ingested more than once then you would have multiple rows with the same ingest id, blob id and other metadata. When we're writing to these tables we want to avoid the cost of doing an UPDATE if a row already exists, so always just write new data to the table

Reply:
I have updated this now to perform the deltion in a single transaction using a 'localTx' (copied from http://scalikejdbc.org/documentation/transaction.html)